File: 83_Remove_Duplicates_from_Sorted_List.java
class Solution {

    public ListNode deleteDuplicates(ListNode head) {
        /* Sanity check */
        if(head == null)
            return head;

        ListNode p = head;
        ListNode q = head;

        while(p != null) {
            /* Move q to the last duplicate */
            while (q.next != null && q.val == q.next.val)
                q = q.next;

            p.next = q.next;
            p = p.next;
            q = q.next;
        }

        return head;
    }

}

File: 83_Remove_Duplicates_from_Sorted_List.cpp
//
//  83. Remove Duplicates from Sorted List.h
//  Project 1
//
//  Created by Jonny Kong on 2/22/16.
//  Copyright © 2016 Jonny Kong. All rights reserved.
//

ListNode* deleteDuplicates(ListNode* head) {
  if (!head || head->next == NULL) return head;

  ListNode* p = head;

  while (p->next) {
    if (p->val == p->next->val)
      p->next = p->next->next;

    else
      p = p->next;
  }

  return head;
}